### ISSUE TITLE
workspace ## output eDP-1 not working

### ISSUE THREAD

**Issue by pypeetpy** [swaywm/sway#5614]
hi, i'm Peet , move from i3 to sway
in i3 we can specify workspace to which monitor we need
in sway i have set 
`workspace 1 output eDP-1
workspace 2 output eDP-1
workspace 3 output eDP-1
workspace 4 output eDP-1
workspace 5 output eDP-1
workspace 6 output eDP-1
workspace 7 output HDMI-1
workspace 8 output HDMI-1
workspace 9 output HDMI-1
workspace 10 output HDMI-1'
in this way but no luck please help
Thankyou

**Comment by RedSoxFan** [comment]
If you look at `swaymsg -t get_outputs`, the output names are likely different than they were under X11. For example, `HDMI-1` is likely `HDMI-A-1`.

**Comment by peet1234** [comment]
> If you look at `swaymsg -t get_outputs`, the output names are likely different than they were under X11. For example, `HDMI-1` is likely `HDMI-A-1`.

yes, i have do that . but setting workspace # output eDP-1 not working

**Comment by RedSoxFan** [comment]
Please post a debug log

**Comment by andrewschott** [comment]
For anyone else having the issue, I was required to quit and restart sway for my ws assignments across my monitors to take effect.  Reloading the config did not work.

HTH someone out.

**Comment by kris7t** [comment]
Workspace assignments in `sway` seem to be cumulative, i.e., if you have assigned a workspace to (or created a workspace on) a certain output, subsequent assignments only take effect if the original output is no longer active. But luckily it's fairly easy to patch sway to overwrite (as opposed to accumulate) assignments: https://git.marussy.com/sway/commit/?h=patch-for/master&id=97111f3e606c7c11dd418c275c3e7c2d69b68a06

**Comment by 71rd** [comment]
this patch seems to still apply cleanly and solves the problem. Is there any reason not to merge it?

**Comment by bl4ckb0ne** [comment]
Nobody bothered to open a pull request
